app/controllers/user/carts_controller.rb

```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+# UserNamespace
+class User
+  # Carts
+  class CartsController < ApplicationController
+    before_action :current_cart
+    def show
+      @cart = @current_cart
```

Comment:
redundant code . make a function and call it in before action for show and destroy